python/variable.cpp

```diff
@@ -128,80 +186,96 @@ void bind_init_0D(py::class_<Variable> &c) {
-
-  // For some reason, pybind11 does not convert python lists to py::array,
-  // so we need to bind the lists manually.
-  // TODO: maybe there is a better way to do this?
-  bind_init_1D<int32_t>(variable);
-  bind_init_1D<double>(variable);
-  // This should be in the certain order
-  bind_init_0D(variable);
-  bind_init_0D<bool>(variable);
-  bind_init_0D<int64_t>(variable);
-  bind_init_0D<double>(variable);
-  bind_init_0D<py::object>(variable);
+
+  binder.append({bind_init_0D<DataArray>, bind::Priority::NativeScipp});
+  binder.append({bind_init_0D<Dataset>, bind::Priority::NativeScipp});
+  binder.append({bind_init_0D<std::string>, bind::Priority::NativeScipp});
+  binder.append({bind_init_0D<Eigen::Vector3d>, bind::Priority::NativeScipp});
+
+  binder.append(
+      {[](py::class_<Variable> &variable) {
+         variable.def(py::init<const VariableProxy &>())
+             .def(py::init(&makeVariableDefaultInit),
+                  py::arg("dims") = std::vector<Dim>{},
+                  py::arg("shape") = std::vector<scipp::index>{},
+                  py::arg("unit") = units::Unit(units::dimensionless),
+                  py::arg("dtype") = py::dtype::of<double>(),
+                  py::arg("variances").noconvert() = false)
+             .def(py::init(&doMakeVariable), py::arg("dims"),
+                  py::arg("values"), // py::array
+                  py::arg("variances") = std::nullopt,
+                  py::arg("unit") = units::Unit(units::dimensionless),
+                  py::arg("dtype") = py::none())
+             .def("rename_dims", &rename_dims<Variable>, py::arg("dims_dict"),
+                  "Rename dimensions.")
+             .def("copy", [](const Variable &self) { return self; },
+                  "Return a (deep) copy.")
+             .def("__copy__", [](Variable &self) { return Variable(self); })
+             .def("__deepcopy__",
+                  [](Variable &self, py::dict) { return Variable(self); })
+             .def_property_readonly("dtype", &Variable::dtype)
+             .def("__radd__", [](Variable &a, double &b) { return a + b; },
+                  py::is_operator())
+             .def("__rsub__", [](Variable &a, double &b) { return b - a; },
+                  py::is_operator())
+             .def("__rmul__", [](Variable &a, double &b) { return a * b; },
+                  py::is_operator())
+             .def("__repr__",
+                  [](const Variable &self) { return to_string(self); });
+       },
+       bind::Priority::NumpyArray});
+
+  binder.append({bind_init_1d_list, bind::Priority::PythonListNamedValues});
+
+  binder.append({bind_init_0D_numpy_types, bind::Priority::NumpyBuffer});
+  binder.append({[](auto &variable) {
+                   bind_init_0D_native_python_types<bool>(variable);
+                   bind_init_0D_native_python_types<int64_t>(variable);
+                   bind_init_0D_native_python_types<double>(variable);
```

Comment:
Does the user of `Binder` still need to take care of ordering types within a priority correctly? If I change the order here, does it still break?

Reply:
No, I've just grouped this wrongly, should introduce the other enum member `PythonNativeBool`.